include/oneapi/dpl/pstl/utils.h

```diff
@@ -155,14 +155,23 @@ class __reorder_pred
     }
 };
 
+class __pstl_assign
+{
+  public:
+    template <typename _Xp, typename _Yp>
+    void
+    operator()(const _Xp& __x, _Yp&& __y) const
+    {
+        __y = __x;
+    }
```

Comment:
Are all the reference semantics correct here?
For an example, look at https://en.cppreference.com/w/cpp/string/char_traits/assign:
```
static constexpr void assign( char_type& r, const char_type& a ) noexcept;
```

Reply:
I think that we fix the case when `std::tuple<_references_>` can be passed. It cannot be `const` lvalue reference because it cannot be assigned to, and it cannot be just lvalue reference because we cannot bind temporary to non-const lvalue reference.

I believe we also need `std::forward<Y>(__y)` but we just would cover corner cases.

Reply:
It's OK to keep rvalue reference then, and use of forward would at least be consistent with that custom assignment brick in this patch.

Reply:
`std::forward` has been added.